### PR TITLE
Constant detail + documentation for completion

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Feature parity with method completion.

Review by commit:

- **Factor out prettyTypeForConstant** (052ba227a)

  Share more code between hover and completion.

  Completion tests have no way to test that the `documentation` is correct
  on a completion item, but hover tests do. Making completion and hover
  share as much code as possible makes it more likely that we'll catch
  bugs when they come up, and fix bugs everywhere rather than isolated to
  one request kind.

- **Constant detail + documentation for completion** (c95d25ee3)

- **Add docs to constants_all_kinds test** (849b902ea)

  Makes sure that the logic that processes the doc comments doesn't crash
  us.

- **Document incorrect behavior for constants docs** (03459ec3e)

### Test plan

This is somewhat hard to test, because there's no framework support for testing
the `detail` nor the `documentation` on completion.

My strategy instead has been to re-use as much logic from **hover** as
possible, and to test hover more thoroughly for constants. That surfaced some
bugs in the existing **hover** support for constants. I've added a test + issue
for those. We can fix that separately. In the cases where docs for constants on
hover now work, so do completion docs.

### Screenshots

Screen Shot 2019-12-06 at 1 59 11 PMScreen Shot 2019-12-06 at 1 59 20 PM
Screen Shot 2019-12-06 at 1 59 28 PM
Screen Shot 2019-12-06 at 1 59 38 PM
Screen Shot 2019-12-06 at 1 59 45 PM
Screen Shot 2019-12-06 at 1 59 54 PM
Screen Shot 2019-12-06 at 2 00 07 PM
Screen Shot 2019-12-06 at 2 00 19 PM
Screen Shot 2019-12-06 at 2 00 30 PM
Screen Shot 2019-12-06 at 2 00 39 PM
Screen Shot 2019-12-06 at 2 00 48 PM

### CODE REVIEW

**Comment by DarkDimius** [comment]
JVilk is the hover-er ;-)

File: main/lsp/lsp_helpers.cc [hunk 1]
```diff
@@ -236,6 +236,15 @@ string prettyTypeForMethod(const core::GlobalState &gs, core::SymbolRef method,
...
 }
 
+string prettyTypeForConstant(const core::GlobalState &gs, core::SymbolRef constant, core::TypePtr type) {
+    core::TypePtr result = type;
+    if (constant.data(gs)->isTypeAlias()) {
+        // By wrapping the type in `MetaType`, it displays as `<Type: Foo>` rather than `Foo`.
+        result = core::make_type<core::MetaType>(type);
+    }
+    return result->showWithMoreInfo(gs);
+}
```
Comment (jez) [inline main/lsp/lsp_helpers.cc]: Honestly not sure what's the point of this snippet. I saw we were doing it for hover, so I decided to do it for completion too. Is it valuable? We could remove it.

File: test/testdata/lsp/hover_constants.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  XXX = nil # wrong
+  # ^ hover: Docs for XXX
+
+  # Docs for CCC
+  CCC = AAA # wrong? arguably showing the dealiased docs is fine
+  # ^ hover: Docs for AAA
+
+  # Docs for EEE
+  EEE = type_member # wrong
+  # ^ hover: Docs for XXX
```
Comment (jvilk-stripe) [inline test/testdata/lsp/hover_constants.rb]: Fix failing test:

```suggestion
  # ^ hover: Docs for EEE
```

File: test/testdata/lsp/hover_constants.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# typed: true
+
+# This file documents incorrect behavior when we findDocumentation for
+# constants via hover.
+#
+# See: https://github.com/sorbet/sorbet/issues/2311
```
Comment (jvilk-stripe) [inline test/testdata/lsp/hover_constants.rb]: The testing framework thinks this is a position assertion

```suggestion
# See https://github.com/sorbet/sorbet/issues/2311
```

File: main/lsp/lsp_helpers.cc (hunk 1 above)
Comment (jvilk-stripe) [reply main/lsp/lsp_helpers.cc]: What snippet? The `MetaType` change?

Comment (jvilk-stripe) [reply main/lsp/lsp_helpers.cc]: iirc it disambiguates from type `Foo` and type `<Type: Foo>`... but I'm willing to accept arguments for different behavior.

**Pull request merged**